sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -487,3 +488,41 @@ def __init__(self):
         self.signed_service = None
         self.signed_version = None
         self.value = None
+
+class TokenAudience():
+    """
+    Base class that defines what calling functions and attributes should be defined for an Azure Active
+    Directory audience token.
+
+    :param str value: The audience to use when requesting tokens for Azure Active Directory authentication.
+        For more information please take a look at:
+        https://learn.microsoft.com/azure/storage/blobs/authorize-access-azure-active-directory. Please use one
+        of the provided classmethods over creating a custom value unless you have a specific scenario for doing so.
+    """
+    DEFAULT_OAUTH_SCOPE = ".default"
+    STORAGE_OAUTH_SCOPE = "https://storage.azure.com/"
+    _value: str
+
+    def __init__(self, value: str) -> None:
+        self._value = value + TokenAudience.DEFAULT_OAUTH_SCOPE
+
+    @classmethod
+    def public_audience(cls) -> Self:
+        """Default Audience. Use to acquire a token for authorizing requests to any Azure Storage account.
+
+        Resource ID: 'https://storage.azure.com/'
+        If no audience is specified, this is the default value.
+
+        :returns: The Default Audience.
+        :rtype: TokenAudience
+        """
+        return cls(TokenAudience.STORAGE_OAUTH_SCOPE)
+
+    def __eq__(self, other: object) -> bool:
+        return self._value == other._value
+
+    def __ne__(self, other: object) -> bool:
+        return self._value != other._value
```

Comment:
Sorry I should have mentioned this earlier but didn't really notice until I saw the API View but do we actually need the `__eq__` and `__ne__` here?

Reply:
Hmm not sure but I did so just to have parity with .NET